java/WordCount.java

```diff
@@ -1,48 +1,50 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
-import java.util.List;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
-import java.util.HashMap;
+import java.util.Map.Entry;
 import java.util.NavigableMap;
+import java.util.StringTokenizer;
 import java.util.TreeMap;
-import java.util.Collections;
-import java.util.regex.Pattern;
 
 /** Word count for Java. Slow because of boxing/unboxing. */
 class WordCount {
     public static void main(String[] args) throws IOException {
         BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
-        Map<String, Integer> m = new HashMap<String, Integer>();
-        Pattern p = Pattern.compile("\\s+");
+        Map<String, Integer> m = new LinkedHashMap<String, Integer>();
```

Comment:
`LinkedHashMap` too, because it maintains the list in addition to the hash table. As I understand, it is faster in iterating through the values, while (obviously) slower at insertion (and I guess deletion). Did it help here? I mean, we insert all elements and then iterate through the map, and that's it, so it can go either way.